drift/lib/src/runtime/query_builder/schema/entities.dart

```diff
@@ -147,10 +147,13 @@ extension NameWithAlias on ResultSetImplementation<dynamic, dynamic> {
   /// can be used in select statements, as it returns something like "users u"
   /// for a table called users that has been aliased as "u".
   String get tableWithAlias {
+    var dialect = attachedDatabase.executor.dialect;
     if (aliasedName == entityName) {
-      return '"$entityName"';
+      return dialect == SqlDialect.mariadb ? '`$entityName`' : '"$entityName"';
```

Comment:
We're using this check in a lot of places. Maybe we should have a `String escapeChar` field on `SqlDialect`? `SqlDialect` could then have a method like this

```dart
String escape(String identifier) => '$escapeChar$identifier$escapeChar';
```

We could then replace usages with `dialect.escape(entityName)` instead of duplicating the check everywhere. In the (unlikely) event that we'll ever have to use a third escape character, that will make it easier to migrate.